### ISSUE TITLE
Implement unions

### ISSUE THREAD

**Issue by diamondburned** [diamondburned/gotk4#25]
Currently, gotk4 does not implement union generation. It can probably be
implemented like so:

```go
type Event struct {
	*event
}

type event struct {
	native *C.GdkEvent
}

// All finalizers omitted for brevity.

func (e *Event) EventType() EventType {
	return *(*EventType)(unsafe.Pointer(e.native))
}

func (e *Event) EventKey() EventKey {
	return EventKey{
		eventKey: &eventKey{
			native: (*C.GdkEventKey)(unsafe.Pointer(e.native)),
		},
	}
}

func Example() {
	var e Event

	switch e.EventType() {
	case gdk.KeyPress:
		e := e.EventKey()
		log.Println("keyval =", e.Keyval())
	}
}
```

Using this is pretty unsafe, but it's probably the cleanest way to implement
this. Type checking is up to the caller to do, because there's no way to do that
with C unions.

**Comment by diamondburned** [comment]
GTK+3 needs events just to handle keys. Without this implemented, that's not
possible.

**Comment by diamondburned** [comment]
Closed by commit d16c0d15.
